rabit/include/rabit/internal/socket.h

```diff
@@ -25,6 +25,14 @@
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 
+#if defined(__sun)
+#include <sys/sockio.h>
+#endif  // defined(__sun)
+
+#if defined(__sun) || defined(sun)
```

Comment:
We need to check both `__sun` and `sun`, according to https://github.com/dmlc/dmlc-core/pull/549

Reply:
Glad that you found the test side!